packages/react/src/radio-group-item/RadioGroupItem.tsx

```diff
@@ -1,37 +1,92 @@
-import * as RadixRadio from "@radix-ui/react-radio-group";
 import { forwardRef, type ReactNode } from "react";
 
 import { Box, type BoxProps } from "../box";
-import { ControlBase } from "../control-base";
+import { CheckboxInput } from "../checkbox-input";
+import { CheckboxInputContent } from "../checkbox-input-content";
+import { CheckboxInputControl } from "../checkbox-input-control";
+import { CheckboxInputDescription } from "../checkbox-input-description";
+import { CheckboxInputHiddenInput } from "../checkbox-input-hidden-input";
+import { CheckboxInputLabel } from "../checkbox-input-label";
+import { useRadioGroupContext } from "../radio-group-context";
+import { extractSprinkles } from "../sprinkles";
 import * as styles from "./RadioGroupItem.css";
 
 type RadioGroupItemProps = BoxProps<
-  typeof RadixRadio.Item,
+  typeof CheckboxInputHiddenInput,
   {
     description?: ReactNode;
   }
 >;
 
-export const RadioGroupItem = forwardRef<HTMLDivElement, RadioGroupItemProps>(
-  ({ children, description, id, value, ...props }, ref) => {
+export const RadioGroupItem = forwardRef<HTMLInputElement, RadioGroupItemProps>(
```

Comment:
todo: rename this to just `Radio` since we can now also use this independently of `RadioGroup`